File: taller-5-java-ronalcastro97/src/main/java/gestion/Zoologico.java
package gestion;
import java.util.ArrayList;
public class Zoologico {
	private String nombre;
	private String ubicacion;
	private ArrayList<Zona>Zona;
	public Zoologico() {
		if(Zona==null)
			Zona=new ArrayList<>();
	}
	
	public Zoologico(String nombre,String ubicacion) {
		this.nombre=nombre;
		this.ubicacion=ubicacion;
		if(Zona==null)
			this.Zona=new ArrayList<>();
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getUbicacion() {
		return ubicacion;
	}

	public void setUbicacion(String ubicacion) {
		this.ubicacion = ubicacion;
	}

	public ArrayList<Zona> getZona() {
		return Zona;
	}

	public void setZona(ArrayList<Zona> zonas) {
		Zona = zonas;
	}
	public void agregarZonas(Zona zona) {
		this.Zona.add(zona);
	}
    public int cantidadTotalAnimales(){
        int total = 0;
        for (int i = 0;i<Zona.size();i++)
            total = total + Zona.get(i).cantidadAnimales();
        
        return total;
    }
}

File: taller-5-python-ronalcastro97/gestion/zoologico.py
class Zoologico():
    _zonas=[]
    def __init__(self,nombre=None, ubicacion=None):
        self._nombre=nombre
        self._ubicacion=ubicacion

    def getNombre(self):
        return self._nombre
    
    def setNombre(self,nombre):
        self._nombre=nombre

    def getUbicacion(self):
        return self._ubicacion

    def setUbicacion(self,ubicacion):
        self._ubicacion=ubicacion

    def agregarZonas(self,zona):
        self._zonas.append(zona)

    def cantidadTotalAnimales(self):
        conta=0
        for zona in self._zonas:
            conta+=zona.cantidadAnimales()
        return conta
    @classmethod
    def getZona(cls):
        return cls._zonas
    @classmethod
    def setZona(cls,zonas):
        cls._zonas= zonas
    


    

